File: 문자열 부수기/BOJ_3448.java
package BOJ_3448;

import java.io.*;

public class BOJ_3448 {

    // https://www.acmicpc.net/problem/3448
    // input
    private static BufferedReader br;

    private static StringBuilder str;
    private static final String EFF = "Efficiency ratio is ";

    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("C:\\Users\\bigyo\\Desktop\\알고리즘\\KotlinAlgo\\src\\main\\kotlin\\BOJ_3448\\res.txt"));
        br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int T = Integer.parseInt(br.readLine());
        while (T-- > 0) {
            str = new StringBuilder();

            while (true) {
                String temp = br.readLine();
                if (temp == null || temp.isEmpty()) {
                    break;
                }
                str.append(temp);
            }

            bw.write(solve());
        }


        bw.close();
    } // End of main()

    private static String solve() {
        StringBuilder sb = new StringBuilder();

        String temp = str.toString();
        double N = temp.length();
        double count = 0.0;

        for (int i = 0; i < N; i++) {
            if (temp.charAt(i) == '#') {
                count++;
            }
        }

        double ratio = 100 - ((count * 100) / N);
        String ratioToString = String.format("%.1f", ratio);
        int ratioToInt = (int) Double.parseDouble(ratioToString);

        sb.append(EFF);
        if (ratioToString.equals(ratioToInt + ".0")) {
            sb.append(ratioToInt);
        } else {
            sb.append(ratioToString);
        }

        sb.append("%.\n");
        return sb.toString();
    } // End of solve()
} // End of Main class

File: 일반구현/BOJ 3448.kt
package BOJ_3448

import java.io.File

// https://www.acmicpc.net/problem/3448
// input
private var br = System.`in`.bufferedReader()

// variables
private lateinit var str: StringBuilder
private const val TEXT = "Efficiency ratio is "

fun main() {
    val path = "C:\\Users\\bigyo\\Desktop\\알고리즘\\KotlinAlgo\\src\\main\\kotlin\\BOJ_3448\\res.txt"
    br = File(path).bufferedReader()
    val bw = System.out.bufferedWriter()

    val T = br.readLine().toInt()

    repeat(T) {
        str = StringBuilder()
        while (true) {
            val temp = br.readLine()
            if (temp.isNullOrEmpty()) break
            str.append(temp)
        }

        bw.write(solve())
    }

    bw.close()
} // End of main()

private fun solve(): String {
    val sb = StringBuilder()

    val target = str.toString()
    val len = target.length.toDouble()
    val count = target.count { it == '#' }.toDouble()

    val temp = (len - count) / len * 100
    val ans = "%.1f".format(temp)
    val ansToInt = ans.toDouble().toInt()


    sb.append(
        "$TEXT${
            if (ans != "$ansToInt.0") {
                "%.1f".format(temp)
            } else {
                "$ansToInt"
            }
        }%.\n"
    )

    return sb.toString()
} // End of solve()